src/colmap/controllers/incremental_mapper.h

```diff
@@ -158,33 +158,50 @@ class IncrementalMapperController : public BaseController {
       std::shared_ptr<const IncrementalMapperOptions> options,
       const std::string& image_path,
       const std::string& database_path,
-      std::shared_ptr<ReconstructionManager> reconstruction_manager);
+      std::shared_ptr<class ReconstructionManager> reconstruction_manager);
 
   void Run();
 
   void TriangulateReconstruction(
       const std::shared_ptr<Reconstruction>& reconstruction);
 
- private:
   bool LoadDatabase();
+
+  // getter functions for python pipelines
+  const std::string ImagePath() const { return image_path_; }
+  const std::string DatabasePath() const { return database_path_; }
```

Comment:
const ref?

Reply:
Above and below for all getters. Or does this not work with Pybind?

Reply:
It should work with pybind. I have updated this.